src/bulkDataRequestCache.ts

```diff
@@ -175,3 +187,106 @@ export class BulkUpsertRequestCache extends BulkDataRequestCache {
     await cache.write();
   }
 }
+
+export class BulkExportRequestCache extends TTLConfig<TTLConfig.Options, BulkExportCacheConfig> {
+  public static getDefaultOptions(): TTLConfig.Options {
+    return {
+      isGlobal: true,
+      isState: true,
+      filename: BulkExportRequestCache.getFileName(),
+      stateFolder: Global.SF_STATE_FOLDER,
+      ttl: Duration.days(7),
+    };
+  }
+
+  public static getFileName(): string {
+    return 'bulk-data-export-cache.json';
+  }
+
+  public static async unset(key: string): Promise<void> {
+    const cache = await BulkUpsertRequestCache.create();
+    cache.unset(key);
+    await cache.write();
+  }
+
+  /**
+   * Creates a new bulk export request cache entry for the given id.
+   */
+  public async createCacheEntryForRequest(
+    bulkRequestId: string,
+    outputInfo: {
+      filePath: string;
+      format: 'csv' | 'json';
+      columnDelimiter: ColumnDelimiterKeys;
+    },
+    username: string | undefined,
+    apiVersion: string | undefined
+  ): Promise<void> {
+    if (!username) {
+      throw messages.createError('usernameRequired');
+    }
+    this.set(bulkRequestId, {
+      jobId: bulkRequestId,
+      outputInfo,
+      username,
+      apiVersion,
+    });
+    await this.write();
+    Logger.childFromRoot('BulkExportCache').debug(`bulk cache saved for ${bulkRequestId}`);
+  }
+
+  public async resolveResumeOptionsFromCache(
+    bulkJobId: string | undefined,
+    useMostRecent: boolean,
+    apiVersion: string | undefined
+  ): Promise<ResumeBulkExportOptions> {
+    if (!useMostRecent && !bulkJobId) {
+      throw messages.createError('bulkRequestIdRequiredWhenNotUsingMostRecent');
+    }
+    const resumeOptionsOptions = {
+      operation: 'query',
+      query: '',
+      pollingOptions: { pollTimeout: 0, pollInterval: 0 },
+    } satisfies Pick<ResumeOptions['options'], 'operation' | 'query' | 'pollingOptions'>;
+
+    if (useMostRecent) {
+      const key = this.getLatestKey();
+      if (!key) {
+        throw messages.createError('cannotFindMostRecentCacheEntry');
```

Comment:
Not as long as the other one, but the name of the error could just be something like, "CacheError" or "MissingCacheEntryError" rather than "CannotFindMostRecentCacheEntryError"